homeassistant/components/ozw/node.py

```diff
@@ -0,0 +1,190 @@
+        """Initialize OZWValidationResponse."""
+        self.success = success
+        self.payload = payload
+        self.err_type = err_type
+        self.err_msg = err_msg
+        self.err_args = args
+
+    @staticmethod
+    def process_fail(err_type, err_msg, *args):
+        """Process an invalid request."""
+        return OZWValidationResponse(False, err_type=err_type, err_msg=err_msg, *args)
+
+    @staticmethod
+    def process_fail_on_type(value, new_value):
+        """Process an invalid request that fails type validation."""
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_SUPPORTED,
+            "Configuration parameter type {} does not match the value type {}",
+            value.type,
+            type(new_value),
+        )
+
+    @staticmethod
+    def process_success(payload):
+        """Process a valid request."""
+        return OZWValidationResponse(True, payload=payload)
+
+
+def set_config_parameter(manager, instance_id, node_id, parameter_index, new_value):
+    """Set config parameter to a node."""
+    instance = manager.get_instance(instance_id)
+    if not instance:
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_FOUND, "OZW Instance not found"
+        )
+
+    node = instance.get_node(node_id)
+    if not node:
+        return OZWValidationResponse.process_fail(ERR_NOT_FOUND, "OZW Node not found")
+
+    value = node.get_value(CommandClass.CONFIGURATION, parameter_index)
+    if not value:
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_FOUND,
+            "Configuration parameter for OZW Node Instance not found",
+        )
+
+    # Bool can be passed in as string or bool
+    if value.type == ValueType.BOOL:
+        if isinstance(new_value, bool):
+            value.send_value(new_value)
+            return OZWValidationResponse.process_success(new_value)
+        if isinstance(new_value, str):
+            if new_value.lower() in ("true", "false"):
+                payload = new_value.lower() == "true"
+                value.send_value(payload)
+                return OZWValidationResponse.process_success(payload)
+
+            return OZWValidationResponse.process_fail(
+                ERR_NOT_SUPPORTED,
+                "Configuration parameter requires true of false",
+            )
+
+        return OZWValidationResponse.process_fail_on_type(value, new_value)
+
+    # List value can be passed in as string or int
+    if value.type == ValueType.LIST:
+        try:
+            new_value = int(new_value)
+        except ValueError:
+            pass
+        if not isinstance(new_value, str) and not isinstance(new_value, int):
+            return OZWValidationResponse.process_fail_on_type(value, new_value)
+
+        for option in value.value["List"]:
+            if new_value not in (option["Label"], option["Value"]):
+                continue
+            payload = int(option["Value"])
```

Comment:
Could this raise if a device isn't following the spec? Should we handle that?

Reply:
If a device isn't following spec in that the `Value` of the list item is not an int, this will indeed raise an exception. I will add a guard here